### ISSUE TITLE
Deleted Save Folder

### ISSUE THREAD

**Issue by BobEngleschmidt** [endless-sky/endless-sky#5211]
I accidently deleted my pilot with 50+ hours of playtime. Even with 20+ snapshots, deleting the pilot deleted them all.

There are no redundancies to protect against this and I have seen others with the same issue.

A simple fix would be to create an additional folder "deleted saves" where any deleted pilots are removed from the active save folder to the inactive backup folder.

I'd be open to other ideas, but as it stands, even using a file recovery software, it is impossible to recover my pilot.

**Comment by pega3** [comment]
There is a very clear security prompt, that explains exactly what is about to happen. If a user ignores that and proceeds, currently the next layer of security is the OS layer. Time machine on Mac, File History or Backup and Restore on Windows, whatever floats your boat on Linux. But in times of regular ransomware attacks, automated and frequent Backups should be the norm.

That said, introducing the suggested feature in a simple implementation (move pilot files into a 'deleted pilots' folder instead of hard-deleting them) would not require much effort. Integrating the recovery of a deleted pilot in the game is a different story and would require and additional panel and dialogues. 

If we go for the simple solution, "delete" could be renamed into "archive" and upon archiving, the instructions on how to recover from the archive (move the file back) could be displayed to the user. This may also add value to players who operate with a lot of pilots and want to clean up their inventory a bit.

**Comment by BobEngleschmidt** [comment]
Archive is a good suggestion. As for the clear prompt, the problem is that after deleting snapshots many times, you get used to clicking through the prompts. But there is both a delete snapshot and a delete pilot option, and when, in a moment of distraction, I clicked the wrong one, I completely ignored the prompt, thinking it was the same again. 
A foolish thing on my part? Definitely. But in googling for a solution I saw that this is definitely not an uncommon issue. The ability to delete all of one's saves at a time is just too risky, without some backup.

    On Friday, August 7, 2020, 12:51:32 AM EDT, pega3 <notifications@github.com> wrote:  
 
 


There is a very clear security prompt, that explains exactly what is about to happen. If a user ignores that and proceeds, currently the next layer of security is the OS layer. Time machine on Mac, File History or Backup and Restore on Windows, whatever floats your boat on Linux. But in times of regular ransomware attacks, automated and frequent Backups should be the norm.

That said, introducing the suggested feature in a simple implementation (move pilot files into a 'deleted pilots' folder instead of hard-deleting them) would not require much effort. Integrating the recovery of a deleted pilot in the game is a different story and would require and additional panel and dialogues.

If we go for the simple solution, "delete" could be renamed into "archive" and upon archiving, the instructions on how to recover from the archive (move the file back) could be displayed to the user. This may also add value to players who operate with a lot of pilots and want to clean up their inventory a bit.

**Comment by Zireael07** [comment]
>  But there is both a delete snapshot and a delete pilot option, and when, in a moment of distraction, I clicked the wrong one, I completely ignored the prompt, thinking it was the same again. 

Are the prompts identical for both? It might be worthwhile to make them clearly different, e.g. bolding out the SNAPSHOT/PILOT part?

**Comment by MCOfficer** [comment]
Never underestimate the stupidity of users. Please don't take offense, that includes myself.

There's a reason many such features have security mechanisms. For example

- Some service i forgot forces you to wait 5 seconds before being able to click OK
- Github & Gitlab require you to type in the repository name



So the way i see it, there are 2 solutions:
1. Force users to read the confirmation prompt (see above)
2. Don't *actually* delete saves, just hide them somehow

**Comment by BobEngleschmidt** [comment]
They are fairly distinct from each other. And deleting it was certainly my stupidity. I would just kick myself and move on, but since I am not the only one who has done the dumb, I figured it would be wise to create a backup of sorts.

    On Friday, August 7, 2020, 08:07:00 AM EDT, Zireael07 <notifications@github.com> wrote:  
 
 



But there is both a delete snapshot and a delete pilot option, and when, in a moment of distraction, I clicked the wrong one, I completely ignored the prompt, thinking it was the same again.


Are the prompts identical for both? It might be worthwhile to make them clearly different, e.g. bolding out the SNAPSHOT/PILOT part?
